pandas/tests/indexes/test_numeric.py

```diff
@@ -678,6 +678,17 @@ def test_constructor_corner(self):
         with tm.assert_raises_regex(TypeError, 'casting'):
             Int64Index(arr_with_floats)
 
+    def test_constructor_overflow_coercion_signed_to_unsigned(self):
+        # GH 15832
+        for t in ['uint8', 'uint16', 'uint32', 'uint64']:
+            with pytest.raises(OverflowError):
+                Index([-1], dtype=t)
+
+    def test_constructor_overflow_coercion_float_to_int(self):
+        # GH 15832
+        with pytest.raises(OverflowError):
```

Comment:
run thru all int & uint dtypes here

I think we have some other tests that this should be grouped with, that are testing the same thing but with NaN's